src/integration-test/java/uk/gov/hmcts/juror/api/moj/report/grouped/CompletionOfServiceReportITest.java

```diff
@@ -0,0 +1,203 @@
+import org.springframework.test.context.jdbc.Sql;
+import uk.gov.hmcts.juror.api.moj.controller.reports.request.StandardReportRequest;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.AbstractReportResponse;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.GroupByResponse;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.GroupedReportResponse;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.GroupedTableData;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.StandardReportResponse;
+import uk.gov.hmcts.juror.api.moj.report.AbstractGroupedReportControllerITest;
+import uk.gov.hmcts.juror.api.moj.report.DataType;
+import uk.gov.hmcts.juror.api.moj.report.ReportHashMap;
+import uk.gov.hmcts.juror.api.moj.report.ReportLinkedMap;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static uk.gov.hmcts.juror.api.TestConstants.COURT_NAME_DISPLAY_NAME;
+import static uk.gov.hmcts.juror.api.TestConstants.COURT_NAME_KEY;
+import static uk.gov.hmcts.juror.api.TestConstants.DATE_FROM_DISPLAY_NAME;
+import static uk.gov.hmcts.juror.api.TestConstants.DATE_FROM_KEY;
+import static uk.gov.hmcts.juror.api.TestConstants.DATE_TO_DISPLAY_NAME;
+import static uk.gov.hmcts.juror.api.TestConstants.DATE_TO_KEY;
+import static uk.gov.hmcts.juror.api.TestConstants.FIRST_NAME_DISPLAY_NAME;
+import static uk.gov.hmcts.juror.api.TestConstants.FIRST_NAME_KEY;
+import static uk.gov.hmcts.juror.api.TestConstants.JUROR_NUMBER_KEY;
+import static uk.gov.hmcts.juror.api.TestConstants.LAST_NAME_DISPLAY_NAME;
+import static uk.gov.hmcts.juror.api.TestConstants.LAST_NAME_KEY;
+import static uk.gov.hmcts.juror.api.TestConstants.VALID_COURT_LOCATION;
+
+@Sql({
+    "/db/truncate.sql",
+    "/db/mod/truncate.sql",
+    "/db/mod/reports/CompletionOfServiceReportITest_typical.sql"
+})
+@SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")//False positive
+class CompletionOfServiceReportITest extends AbstractGroupedReportControllerITest {
+
+    public static final String COMPLETION_DATE_KEY = "completion_date";
+
+    @Autowired
+    public CompletionOfServiceReportITest(TestRestTemplate template) {
+        super(template, CompletionOfServiceReport.class);
+    }
+
+    @Override
+    protected String getValidJwt() {
+        return getCourtJwt(VALID_COURT_LOCATION);
+    }
+
+    @Override
+    protected StandardReportRequest getValidPayload() {
+        return addReportType(StandardReportRequest.builder()
+            .fromDate(LocalDate.of(2023, 1, 1))
+            .toDate(LocalDate.of(2023, 2, 1))
+            .build());
+    }
+
+    @Test
+    void positiveTypicalCourt() {
+        testBuilder()
+            .triggerValid()
+            .responseConsumer(this::verifyAndRemoveReportCreated)
+            .assertEquals(createResponse("CHESTER (415)", 7,
+                true));
+    }
+
+    @Test
+    @SuppressWarnings({
+        "PMD.JUnitTestsShouldIncludeAssert"//False positive
+    })
```

Comment:
Not needed should be at top of class

Reply:
Removed (instance at top of class retained).